### DISCUSSION TITLE
Introducing CopilotForXcodeKit

### DISCUSSION THREAD

**Discussion by intitni** [intitni/CopilotForXcode#405] [Announcements]
Now you can start to build extensions for Copilot for Xcode. The first supported extension type is the suggestion service. Support for custom chat tabs, chat service, etc, will come later.

To build an extension, you need to create a generic extension target with the help of this package [CopilotForXcodeKit](https://github.com/intitni/CopilotForXcodeKit). And your app can remain sandboxed!

Download the preview version that supports extensions [here (0.28.4)](https://github.com/intitni/CopilotForXcode/releases/tag/0.28.4).

I also plan to migrate GitHub Copilot and Codeium support to use extensions in the coming year. It will make them easier to maintain by the community and will make it easier to support their chat services (if possible) through either a custom chat service or a custom chat tab.

I will also make an extension that provides suggestion service via custom models.

If you are interested in building an extension for Copilot for Xcode, please let me know what APIs you would like to have!

<img width="917" alt="Screenshot 2023-12-30 at 22 11 00" src="https://github.com/intitni/CopilotForXcode/assets/793147/15592479-7154-4853-bf40-de1957dd7ebd">

**Comment by elanchou** [comment]
Nice work

**Comment by timglenn123** [comment]
api for appstore connect? update testflight metadata and builds, review, stop workflow builds without having to log into appstore connect.
